packages/common/driver-definitions/src/storage.ts

```diff
@@ -101,9 +102,16 @@ export interface IDocumentStorageServicePolicies {
     readonly minBlobSize?: number;
 
     /**
+     * @deprecated Use snapshotCacheDisabledForTesting instead - Only enabling/disabling will be exposed
      * This policy tells the runtime that the driver will not use cached snapshots older than this value.
      */
     readonly maximumCacheDurationMs?: number;
+
+    /**
+     * If true, the driver will not use the snapshot cache but will always fetch from the service. Defaults to false.
+     * This configuration is only expected to be used for test scenarios.
+     */
+    readonly snapshotCacheDisabledForTesting?: boolean;
```

Comment:
I don't think we should expose this option here in a public interface. We don't want this to be a policy that hosts can set.
The idea behind adding `maximumCacheDurationMs` as a policy was this:
A host can choose to expire their snapshots very quickly in order to be up-to-date. However, they cannot set this value to be greater than 2 days, only smaller. This gives GC an upper bound on the time it needs to wait for sweeping objects.

A proposal of what we can do - Either have `maximumCacheDurationMs` or remove it. Irrespective of that, the maximum duration of 2 days is fixed and enforeced and that is what is communicated to the runtime and GC. If we want to override this value, it can be done via feature flags and that value is communicated to the runtime by the driver.

Reply:
Note that ODSP does not expose these policies today but other drivers do and they can be set by hosts.

Reply:
@vladsud I would love your thoughts on this since I don't have a lot of experience with driver / host poclicies.

Reply:
Biggest question from this thread is what policies are implemented by host v. Driver. This policy type is not passed in as a means to control the driver but is exposed by the driver to enable proper interop.

Reply:
> If we want to override this value, it can be done via feature flags and that value is communicated to the runtime by the driver.

This is exactly what I'm doing, except I'm using a boolean feature flag not a number, to cement the fact that this 2-day value really isn't configurable.

Maybe we should look through the code together to see if I'm missing something.

Reply:
To expand on my previous comment about driver v. host policies:

> I don't think we should expose this option here in a public interface. We don't want this to be a policy that hosts can set.

This policy is set by the driver, not passed in as a means to control the driver. The driver exposes this info as FYI to other layers to enable proper interop.

> The idea behind adding maximumCacheDurationMs as a policy was this:
> A host can choose to expire their snapshots very quickly in order to be up-to-date. However, they cannot set this value to be greater than 2 days, only smaller. 

I wouldn't say it like that.  Hosts can certainly implement their cache to keep older than 2 days.  But the _driver_ declares it will not use cache entries older than this policy - i.e. 2 days.

The problem was, this value was not plumbed through to the place that implemented it.  See #11168.

The change in this PR is to change the language the driver uses to talk about this.  Instead of declaring how old of snapshots may be used, that info is hardcoded throughout the driver, in accordance with the documented requirement of GC.  However, I do want the driver to own whether snapshot cache is disabled altogether, because we need this capability to test GC without waiting for days.

Alternative is to get rid of this policy, and have the Runtime use the same feature flag name, but I would rather that all live within the driver - especially when thinking of other drivers and how they would be implemented.

Reply:
> This policy is set by the driver, not passed in as a means to control the driver. The driver exposes this info as FYI to other layers to enable proper interop.
> 

This is true only for ODSP today. Other drivers do allow you to provide this value during creation - https://github.com/microsoft/FluidFramework/blob/215770d4613df9c3660eea24bb490cd1ed002aec/packages/drivers/routerlicious-driver/src/documentStorageService.ts#L73.

Reply:
> > If we want to override this value, it can be done via feature flags and that value is communicated to the runtime by the driver.
> 
> This is exactly what I'm doing, except I'm using a boolean feature flag not a number, to cement the fact that this 2-day value really isn't configurable.
> 
> Maybe we should look through the code together to see if I'm missing something.

What I was trying to say is do this via feature flags without exposing is as a policy.

Reply:
> This is true only for ODSP today. Other drivers do allow you to provide this value during creation 

Oh... good to know, thanks for pointing it out. I'll update the comment I added to that interface in this PR.

> What I was trying to say is do this via feature flags without exposing is as a policy.

This would mean we'd have the ODSP driver and runtime reading the same feature flag and depending on that fact.  Other drivers would not necessarily respect that same flag.  It's a pretty undesirable and brittle coupling.  Besides that or this policy, I don't know how to plumb the data through.

In light of what you pointed out about how this policy interface may be used, I agree it's a little strange to put on there, but I prefer that approach to trying to keep feature flag names / semantics in touch across all driver implementations + runtime.

Reply:
@agarwal-navin are you able to explain again your concern with changing the policy? It doesn't feel disruptive to do so, but it seems like it does to you. Or risky somehow?

Reply:
Another idea which started as a joke but now I'm thinking about it - what about changing the type of the existing policy to 0 | 2 days (numerically)? That will document the 2 day assumption/requirement but also allow for test override.

Reply:
My concern is that we should be careful about policies since these are meant to be provided by the hosts to the driver. We should not add test specific policies for example. Since we are updating policies here, we should be confident that we are adding the right one and not make the same mistake we did with the existing policy for cache expiry.

Reply:
> Another idea which started as a joke but now I'm thinking about it - what about changing the type of the existing policy to 0 | 2 days (numerically)? That will document the 2 day assumption/requirement but also allow for test override.

This seems similar to making this a boolean and having the cache expiry value as 2 when the policy is true :-). The current policy also works for testing by setting it to 0 right? I am just trying to understand the value of changing this w.r.t. what currently exists (which I agree is not ideal).

Reply:
Value is not having to plumb the number throughout the ODSP driver. Can just read the feature flag both places.

Reply:
And less surface area in GC to think about differing values over lifetime of a file. 0 is clearly about testing so more lenient to closing the container of it doesn't match, or just letting it be mismatched

Reply:
That sounds good to me.

Reply:
Ok. Let's sleep on it. And @vladsud if you can make your way through this thread and give input I'd like to hear it. 

Downside to this plan is it's a breaking change, but practically it's fine for the next few weeks to stage it.

Reply:
For context for anyone following along, the reason the existing approach is challenging is (A) it's hard with current code to plumb the same value to both epochTracker and Runtime - but that is solvable - and (B) GC would require the value to never change. So it needs to be persisted in the file. But there's no good way to do that as-is. So the short-term plan is to lock it at 2 days in the code until we get to the long term plan which is to move the code that retrieves snapshots from the cache into the runtime layer. Then we can persist the policy in the file and reject anything from the cache violating that policy. Need more specific design, and need to get it on the backlog. Not an urgent priority in that it doesn't block roll-out of GC to Bohemia.

Reply:
#12183